.travis.yml

```diff
@@ -42,17 +40,31 @@ install:
   - conda config --set always_yes yes --set changeps1 no
   - conda install conda-build anaconda-client
   - conda update -q conda conda-build
-  - conda config --add channels $PYDM_CHANNEL 
   - conda config --append channels pcds-tag
   - conda config --append channels conda-forge
   # Useful for debugging any issues with conda
   - conda info -a
-  # Grab all dependencies
-  - conda build -q conda-recipe --python=$TRAVIS_PYTHON_VERSION --output-folder bld-dir
-  - conda config --add channels "file://`pwd`/bld-dir"
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION typhon --file dev-requirements.txt
-  # Launch Conda environment
-  - source activate test-environment
+  - |
+    # Install all dependencies via CONDA
+    if [[ $PYDM_CHANNEL != "None" ]]; then
+      conda config --add channels $PYDM_CHANNEL
+      # Use the recipe contained inside the repository to build packcage
+      conda build -q conda-recipe --python=$TRAVIS_PYTHON_VERSION --output-folder bld-dir
+      # Install package with development dependencies
+      conda config --add channels "file://`pwd`/bld-dir"
+      conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION typhon --file dev-requirements.txt
+      # Launch Conda environment
+      source activate test-environment
+    # Install all dependencies via PIP
+    else
+      conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION pip
+      # Launch Conda environment
+      source activate test-environment
+      # Install package
+      python setup.py install
+      # Add our dependencies
+      pip install git+https://github.com/pcdshub/happi.git PyQt5 -r dev-requirements.txt
```

Comment:
It looks like you're always testing with happi installed, but not explicitly requiring it for the pip install. Maybe there should be a test env that runs all non-happi tests without happi to make sure it's still ok to omit it.

Reply:
Fair point. I have this test which checks that everything is importable without `happi` but it might be worth adding a branch here to really test it. With the CONDA install it was awkward because it is listed as a hard requirement in the CONDA recipe so I would have had to uninstall it. However, now that I'm tacking it on to the dev-requirements with pip it would be pretty trivial to not add

https://github.com/pcdshub/typhon/blob/67afff27833471b11cfca6998066ed9b64329ec5/tests/plugins/test_happi.py#L52